src/quantum_gates/_simulation/circuit.py

```diff
@@ -594,6 +594,269 @@ def reset(self):
         self.phi = [0 for i in range(self.nqubit)]
 
 
+class BinaryCircuit(object):
```

Comment:
@paolodr98 Could you explain the difference between the different (if any) circuit classes in the next meeting? Maybe we could combine them somehow and remove some redundancy. Now it is a lot of similar code.